pkg/datapath/iptables/iptables.go

```diff
@@ -694,6 +695,31 @@ func (m *IptablesManager) InstallProxyRules(proxyPort uint16, ingress bool, name
 	return m.iptProxyRules("-A", proxyPort, ingress, name)
 }
 
+// GetProxyPort finds a proxy port used for redirect 'name' installed earlier with InstallProxyRules.
+// By convention "ingress" or "egress" is part of 'name' so it does not need to be specified explicitly.
+// Returns 0 a TPROXY entry with 'name' can not be found.
+func (m *IptablesManager) GetProxyPort(name string) uint16 {
+	prog := "iptables"
+	if !option.Config.EnableIPv4 {
+		prog = "ip6tables"
+	}
+
+	res, err := runProgCombinedOutput(prog, []string{"-t", "mangle", "-n", "-L", ciliumPreMangleChain}, false)
+	if err != nil {
+		return 0
+	}
+
+	re := regexp.MustCompile(name + ".*TPROXY redirect 0.0.0.0:([1-9][0-9]*) mark")
+	str := re.FindString(string(res))
+	portStr := re.ReplaceAllString(str, "$1")
+	portInt, err := strconv.Atoi(portStr)
+	if err != nil {
+		log.WithError(err).Debugf("Port number cannot be parsed: %s", portStr)
+		return 0
+	}
+	return uint16(portInt)
+}
```

Comment:
Hmm, this should be good for v1.8 but for latest with tproxy it will not work. 

Probably makes sense to merge the rest of this PR as-is and have a separate commit to address the tproxy case. Not sure whether latest code will pass with this broken (maybe it'll pass if we don't have an actual runtime test for this condition..)

Reply:
@joestringer Any hints how to do this with the new bpf tproxy? Is there a header file left over by the previous run we could parse instead?

Reply:
AFAIK the port is only placed into BPF policy maps for each endpoint, which isn't the most reliable way to store/restore proxy ports for all protocols.

I think we could consider writing each proxy port to one of the config files, purely for restore purposes.

Reply:
I've filed https://github.com/cilium/cilium/issues/13208 to follow up.